warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -1503,7 +1538,7 @@ public boolean isCancelledQuery() {
                     synchronized (this) {
                         // now recheck the cancelled flag and timeout to ensure we really need to make the hdfs calls
                         if (!cancelled && ((now - lastCancelledCheck) > CANCELLED_CHECK_INTERVAL)) {
-                            cancelled = !queryLock.isQueryRunning();
+                            cancelled |= !queryLock.isQueryRunning();
```

Comment:
if you get to this statement, `cancelled` is already known to be false.  no need for `|=` in that case.

Reply:
woops